geode-core/src/distributedTest/java/org/apache/geode/management/JMXMBeanReconnectDUnitTest.java

```diff
@@ -273,18 +321,57 @@ private static MBeanServerConnection connectToMBeanServer(String url) throws IOE
         .collect(toList());
   }
 
-  private void waitForLocatorsToAgreeOnMembership() {
-    await()
-        .until(
-            () -> {
-              int locator1BeanCount =
-                  getFederatedGemfireBeansFrom(locator1)
-                      .size();
-              int locator2BeanCount =
-                  getFederatedGemfireBeansFrom(locator2)
-                      .size();
-              return locator1BeanCount == locator2BeanCount;
-            });
+  /**
+   * Waits until the two locators have the same MBeans, and have NUM_BEANS beans.
+   *
+   * This method will return when the locators have the same beans and the expected number of beans.
+   * If it does not complete within GeodeAwaitility.DEFAULT_TIMEOUT seconds, a TimeoutException will
+   * be thrown.
+   */
+  private void waitForLocatorsToAgreeOnMembership()
+      throws InterruptedException, IOException {
+    List<ObjectName> l1Beans = new ArrayList<>();
+    List<ObjectName> l2Beans = new ArrayList<>();
+    AtomicBoolean consistent = new AtomicBoolean();
+
+    Stopwatch stopwatch = Stopwatch.createStarted();
+    do {
+      // reset everything
+      consistent.set(true);
+      l1Beans.clear();
+      l2Beans.clear();
+
+      // get the MBeans and add them to lists to check them
+      l1Beans.addAll(getFederatedGemfireBeansFrom(locator1Connection));
+      l2Beans.addAll(getFederatedGemfireBeansFrom(locator2Connection));
+
+      if (stopwatch.elapsed(TimeUnit.SECONDS) >= TIMEOUT) {
```

Comment:
Are we expecting above "getFederatedGemfireBeansFrom" calls to be returned before the timeout? Why are we failing here.